### PR TITLE
9530 Fixed Have Changes check

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#9530

*Description of changes:*
- the Have Changes check was not prompting when Cancel was clicked
- navigating away from the page was triggering even after cancelling (no more outstanding changes)
- fixed for AR, COA, COD and CORR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/AnnualReport.vue [hunk 1]
```diff
@@ -608,36 +608,6 @@ export default class AnnualReport extends Mixins(
     this.$root.$off('fetch-error-event')
   }
 
-  beforeRouteLeave (to, from, next): void {
```
Comment (severinbeauvais) [inline src/views/AnnualReport.vue:611]: I didn't see this lifecycle event being called. It used to work. 🤷‍♂️ 

This code is now in `goToDashboard()`.

File: src/views/StandaloneOfficeAddressFiling.vue [hunk 2]
```diff
@@ -757,11 +725,41 @@ export default class StandaloneOfficeAddressFiling extends Mixins(
...
+      // route to dashboard
+      this.$router.push({ name: Routes.DASHBOARD })
+        .catch(() => {}) // ignore error in case navigation was aborted
+      return
+    }
+
+    // open confirmation dialog and wait for response
+    this.$refs.confirm.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Change of Directors. Do you want to exit your filing?',
```
Comment (argush3) [inline src/views/StandaloneOfficeAddressFiling.vue:744]: should text be something else here?  i.e. something related to office address?

Comment (severinbeauvais) [reply src/views/StandaloneOfficeAddressFiling.vue:744]: Oops !!! Fixing this now...

File: src/views/Correction.vue [hunk 3]
```diff
@@ -741,11 +708,41 @@ export default class Correction extends Mixins(
...
+      // route to dashboard
+      this.$router.push({ name: Routes.DASHBOARD })
+        .catch(() => {}) // ignore error in case navigation was aborted
+      return
+    }
+
+    // open confirmation dialog and wait for response
+    this.$refs.confirm.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Change of Directors. Do you want to exit your filing?',
```
Comment (argush3) [inline src/views/Correction.vue:727]: is text correct here?

Comment (severinbeauvais) [reply src/views/Correction.vue:727]: Fixing this too. And also Annual Report.

File: src/views/AnnualReport.vue [hunk 4]
```diff
@@ -1107,11 +1075,41 @@ export default class AnnualReport extends Mixins(
...
+      // route to dashboard
+      this.$router.push({ name: Routes.DASHBOARD })
+        .catch(() => {}) // ignore error in case navigation was aborted
+      return
+    }
+
+    // open confirmation dialog and wait for response
+    this.$refs.confirm.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Change of Directors. Do you want to exit your filing?',
```
Comment (argush3) [inline src/views/AnnualReport.vue:1094]: text update req'd?

Comment (severinbeauvais) [reply src/views/AnnualReport.vue]: Yes, all 3 fixed now.

**Review by argush3 — APPROVED** [review]

**Pull request merged**